osbs/api.py

```diff
@@ -529,6 +530,15 @@ def get_token(self):
         return self.os.get_oauth_token()
 
     @osbsapi
+    def login(self, token):
+        token_file = utils.get_instance_token_file_name(self.os_conf.conf_section)
+        token_file_dir = os.path.dirname(token_file)
+        if not os.path.exists(token_file_dir):
+            os.makedirs(token_file_dir)
+        with open(token_file, 'w') as f:
+            f.write(token)
```

Comment:
We can't create the file and set its permissions afterwards; we have to do it in one step (otherwise someone could open the file in the interim and hold onto the file descriptor).

I think you can only do this in Python with `os.open`.